warehouse/admin/views/blacklist.py

```diff
@@ -0,0 +1,251 @@
+            )
+
+        blacklist_query = blacklist_query.filter(or_(*filters))
+
+    blacklist = SQLAlchemyORMPage(
+        blacklist_query,
+        page=page_num,
+        items_per_page=25,
+        url_maker=paginate_url_factory(request),
+    )
+
+    return {"blacklist": blacklist, "query": q}
+
+
+@view_config(
+    route_name="admin.blacklist.add",
+    renderer="admin/blacklist/confirm.html",
+    permission="admin",
+    request_method="GET",
+    uses_session=True,
+)
+def confirm_blacklist(request):
+    project_name = request.GET.get("project")
+    if project_name is None:
+        raise HTTPBadRequest("Must have a project to confirm.")
+
+    comment = request.GET.get("comment", "")
+
+    # We need to look up to see if there is an existing project, releases,
+    # files, roles, etc for what we're attempting to blacklist. If there is we
+    # need to warn that blacklisting will delete those.
+    project = (
+        request.db.query(Project)
+                  .filter(Project.normalized_name ==
+                          func.normalize_pep426_name(project_name))
+                  .first()
+    )
+    if project is not None:
+        releases = (
+            request.db.query(Release)
+                      .filter(Release.name == project.name)
+                      .all()
+        )
+        files = (
+            request.db.query(File)
+                      .filter(File.name == project.name)
+                      .all()
+        )
+        users = [
+            r.user
+            for r in (
+                request.db.query(Role)
+                .join(User)
+                .filter(Role.project == project)
+                .distinct(User.username)
+                .order_by(User.username)
+                .all()
+            )
+        ]
+    else:
+        releases = []
+        files = []
+        users = []
+
+    return {
+        "blacklist": {
+            "project": project_name,
+            "comment": comment,
+        },
+        "existing": {
+            "project": project,
+            "releases": releases,
+            "files": files,
+            "users": users,
+        }
+    }
+
+
+@view_config(
+    route_name="admin.blacklist.add",
+    permission="admin",
+    request_method="POST",
+    uses_session=True,
+    require_methods=False,
+)
+def add_blacklist(request):
+    project_name = request.POST.get("project")
+    if project_name is None:
+        raise HTTPBadRequest("Must have a project to confirm.")
+    comment = request.POST.get("comment", "")
+
+    # Verify that the user has confirmed the request to blacklist.
+    confirm = request.POST.get("confirm")
+    if not confirm:
+        request.session.flash(
+            "Must confirm the blacklist request.",
+            queue="error",
+        )
+        return HTTPMovedPermanently(request.current_route_path())
+    elif canonicalize_name(confirm) != canonicalize_name(project_name):
+        request.session.flash(
```

Comment:
the flash produced is pretty much identical in color to the header in admin, not a huge deal but threw me off when reviewing. 
<img width="605" alt="screen shot 2017-09-16 at 7 34 29 pm" src="https://user-images.githubusercontent.com/1200832/30516759-20104d4c-9b16-11e7-9f92-76967874e5cb.png">